enterprise/models.py

```diff
@@ -2461,6 +2461,103 @@ class LicensedEnterpriseCourseEnrollment(EnterpriseFulfillmentSource):
     )
 
 
+class DefaultEnterpriseEnrollmentIntention(TimeStampedModel, SoftDeletableModel):
+    """
+    Specific to an enterprise customer, this model defines a course or course run
+    that should be auto-enrolled for any enterprise customer user linked to the customer.
+
+    .. no_pii:
+    """
+    DEFAULT_ENROLLMENT_CONTENT_TYPE_CHOICES = [
+        ('course', 'Course'),
+        ('course_run', 'Course Run'),
+    ]
+    uuid = models.UUIDField(
+        primary_key=True,
+        default=uuid4,
+        editable=False,
+    )
+    enterprise_customer = models.ForeignKey(
+        EnterpriseCustomer,
+        blank=False,
+        null=False,
+        related_name="default_course_enrollments",
+        on_delete=models.deletion.CASCADE,
+        help_text=_(
+            "The customer for which this default enrollment will be realized.",
+        )
+    )
+    content_type = models.CharField(
```

Comment:
[curious] What's your thinking around how this field would be utilized? From a functionality perspective, I'm not we'd necessary need it per se for the computed properties to function BUT having this field would definitely help better understand/analyze/query these models to understand how many courses vs. course runs are configured.

Do you think this is intended to be manually set at time of configuring a `DefaultEnterpriseEnrollmentIntention`, or would it be programmatically set after saving based on deducing whether the provided `content_key` is a course vs. course run (perhaps based on the content metadata API request made to enterprise-catalog which I believe returns the `content_key` vs. `parent_content_key` relationship?

Reply:
Either one - we could populate it automatically based on a simple regular expression.